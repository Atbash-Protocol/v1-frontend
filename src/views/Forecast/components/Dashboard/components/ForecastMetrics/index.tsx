import { Grid } from '@mui/material';
import Decimal from 'decimal.js';
import { t } from 'i18next';

import MenuMetric from 'components/Metrics/MenuMetric';
import { formatUSDFromDecimal } from 'helpers/price-units';

interface ForecastMetricsProps {
    BASHPrice: Decimal;
    SBASHBalance: Decimal;
    stakingPercentage: Decimal;
}

const ForecastMetrics = ({ BASHPrice, stakingPercentage, SBASHBalance }: ForecastMetricsProps) => {
    const metrics = [
        { key: 'BASHPrice', value: formatUSDFromDecimal(BASHPrice, 2) },
        { key: 'globe:CurrentRewardYield', value: formatUSDFromDecimal(stakingPercentage) },
        { key: 'globe:YourStakedBASHBalance', value: `${SBASHBalance.toFixed(2)} SBASH` },
    ].map(({ key, value }) => (
        <Grid item key={key} xs={6} sm={4} md={4} lg={4} mt={2}>
            <MenuMetric metricKey={t(key)} value={value} />
        </Grid>
    ));

    return (
        <Grid container spacing={2}>
            {metrics}
        </Grid>
    );
};

export default ForecastMetrics;
