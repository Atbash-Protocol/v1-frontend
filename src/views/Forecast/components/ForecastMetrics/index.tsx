import { Grid } from '@mui/material';
import { t } from 'i18next';
import { useSelector } from 'react-redux';

import MenuMetric from 'components/Metrics/MenuMetric';
import { selectSBASHBalance } from 'store/modules/account/account.selectors';
import { selectFormattedReservePrice } from 'store/modules/app/app.selectors';
import { selectStakingRebasePercentage } from 'store/modules/metrics/metrics.selectors';

const ForecastMetrics = () => {
    const BASHPrice = useSelector(selectFormattedReservePrice);
    const stakingPercentage = useSelector(selectStakingRebasePercentage);
    const SBashBalance = useSelector(selectSBASHBalance);

    const metrics = [
        { key: 'BASHPrice', value: BASHPrice },
        { key: 'globe:CurrentRewardYield', value: stakingPercentage },
        { key: 'globe:YourStakedBASHBalance', value: `${SBashBalance} SBASH` },
    ].map(({ key, value }) => (
        <Grid item key={key} xs={6} sm={4} md={4} lg={4} mt={2}>
            <MenuMetric metricKey={t(key)} value={value} />
        </Grid>
    ));

    return (
        <Grid container spacing={2}>
            {metrics}
        </Grid>
    );
};

export default ForecastMetrics;
