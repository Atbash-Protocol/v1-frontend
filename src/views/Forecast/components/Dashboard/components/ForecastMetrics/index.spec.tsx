import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import Decimal from 'decimal.js';
import { Provider } from 'react-redux';

import ForecastMetrics from '.';

jest.mock('components/Metrics/MenuMetric', () => (props: any) => <> {JSON.stringify(props)}</>);

const renderComponent = (component: JSX.Element) => render(<Provider store={configureStore({ reducer: jest.fn() })}>{component}</Provider>);

describe('Metrics', () => {
    it('renders correctly', () => {
        const props = {
            BASHPrice: new Decimal(10),
            SBASHBalance: new Decimal(50),
            stakingPercentage: new Decimal(12.293),
        };

        const component = renderComponent(<ForecastMetrics {...props} />);

        expect(component).toMatchSnapshot();
    });
});
