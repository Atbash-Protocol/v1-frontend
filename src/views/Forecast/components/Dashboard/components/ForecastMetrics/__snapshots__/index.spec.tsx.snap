// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
        >
           
          {"value":"$10.00"}
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
        >
           
          {"value":"$12"}
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
        >
           
          {"value":"50.00 SBASH"}
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
      >
         
        {"value":"$10.00"}
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
      >
         
        {"value":"$12"}
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 css-17sqnp1-MuiGrid-root"
      >
         
        {"value":"50.00 SBASH"}
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
